hathor/verification/transaction_verifier.py

```diff
@@ -47,10 +49,84 @@ def verify(self, tx: Transaction, *, reject_locked_reward: bool = True) -> None:
         if tx.is_genesis:
             # TODO do genesis validation
             return
+        self.verify_without_storage(tx)
+        self.verify_sigops_input(tx)
+        self.verify_inputs(tx)  # need to run verify_inputs first to check if all inputs exist
+        self.verify_parents(tx)
+        self.verify_sum(tx)
+        if reject_locked_reward:
+            self.verify_reward_locked(tx)
+
+    def verify_unsigned_skip_pow(self, tx: Transaction) -> None:
+        """ Same as .verify but skipping pow and signature verification."""
+        tx.verify_unsigned_skip_pow()
+
+    @staticmethod
+    def verify_parents_basic(tx: Transaction) -> None:
+        """Verify number and non-duplicity of parents."""
+        tx.verify_parents_basic()
+
+    def verify_weight(self, tx: Transaction) -> None:
+        """Validate minimum tx difficulty."""
+        tx.verify_weight()
+
+    def verify_without_storage(self, tx: Transaction) -> None:
+        """ Run all verifications that do not need a storage.
+        """
         tx.verify_without_storage()
+
+    def verify_sigops_input(self, tx: Transaction) -> None:
+        """ Count sig operations on all inputs and verify that the total sum is below the limit
+        """
         tx.verify_sigops_input()
-        tx.verify_inputs()  # need to run verify_inputs first to check if all inputs exist
-        tx.verify_parents()
+
+    def verify_inputs(self, tx: Transaction, *, skip_script: bool = False) -> None:
+        """Verify inputs signatures and ownership and all inputs actually exist"""
+        tx.verify_inputs(skip_script=skip_script)
+
+    @staticmethod
+    def verify_script(*, tx: Transaction, input_tx: TxInput, spent_tx: BaseTransaction) -> None:
+        """
+        :type tx: Transaction
+        :type input_tx: TxInput
+        :type spent_tx: Transaction
+        """
+        tx.verify_script(input_tx, spent_tx)
+
+    def verify_sum(self, tx: Transaction) -> None:
+        """Verify that the sum of outputs is equal of the sum of inputs, for each token.
+
+        If there are authority UTXOs involved, tokens can be minted or melted, so the above rule may
+        not be respected.
+
+        :raises InvalidToken: when there's an error in token operations
+        :raises InputOutputMismatch: if sum of inputs is not equal to outputs and there's no mint/melt
+        """
         tx.verify_sum()
-        if reject_locked_reward:
-            tx.verify_reward_locked()
+
+    @staticmethod
+    def verify_reward_locked(tx: Transaction) -> None:
```

Comment:
Why is it a static method?

Reply:
Changed in https://github.com/HathorNetwork/hathor-core/pull/797/commits/105e0598363e0125e9f6112ec72b213889790b61

Reply:
Thanks!